lib/ceedling/configurator.rb

```diff
@@ -193,17 +194,34 @@ def find_and_merge_plugins(config)
     end
 
     plugin_yml_defaults.each do |defaults|
-      @configurator_builder.populate_defaults( config, @yaml_wrapper.load(defaults) )
+      default_config.deep_merge!( @yaml_wrapper.load(defaults) )
     end
 
     plugin_hash_defaults.each do |defaults|
-      @configurator_builder.populate_defaults( config, defaults )
+      @configurator_builder.populate_defaults( default_config, defaults )
     end
 
     # special plugin setting for results printing
     config[:plugins][:display_raw_test_results] = true if (config[:plugins][:display_raw_test_results].nil?)
 
     paths_hash.each_pair { |name, path| config[:plugins][name] = path }
+    paths_hash.each_pair { |name, path| default_config[:plugins][name] = path } if (config[:plugins].nil?)
```

Comment:
This doesn't look right to me, if `config[:plugins].nil?` then won't the previous line crash trying to access `config[:plugins][name]`?

Reply:
You are right, the condition is wrong. The condition should verify if the :plugins key exists in default_config, as :plugin has been relocated from DEFAULT_CEEDLING_CONFIG to CEEDLING_CONFIG_INTERNAL.

Reply:
I think we can delete this line, no need to add the plugin paths to the default config